components/constellation/constellation.rs

```diff
@@ -1657,14 +1653,20 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         let (browsing_context_id, parent_info) = match self.pipelines.get(&source_id) {
             Some(pipeline) => (pipeline.browsing_context_id, pipeline.parent_info),
             None => {
-                warn!("Pipeline {:?} loaded after closure.", source_id);
+                warn!("Pipeline {} loaded after closure.", source_id);
+                return None;
+            }
+        };
+        let (top_level_id, window_size, timestamp) = match self.browsing_contexts.get(&browsing_context_id) {
```

Comment:
These values are only used in the `None` branch of the match, can we move them there?